### PR TITLE
CVLデータセットを学習させる

### PR DESCRIPTION
**Pull request by matsumishoki**
CVLデータセットを学習させる

### CODE REVIEW

File: convnet.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        y_select_point = np.random.permutation(y_select_points)
+        y_p = y_select_point[0]
+        x_p = x_select_point[0]
+
+        image = image[y_p:y_p+image_size, x_p:x_p+image_size]
+        plt.imshow(image, cmap=plt.cm.gray)
+        plt.show()
+        plt.draw()
+        images.append(image)
+        x = np.array(images).reshape(-1, 1, 200, 200)
```
Comment (ronekko) [inline convnet.py]: これはforを抜けた後で．
また，200→image_size

File: convnet.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+# エポック毎に訓練データを作成する関数
+def make_epoch_data():
+    # 検索するデータセットのファイルのtop_pathを指定する
+    top_path = os.path.join("CVL_ConvNet_data")
+    temp_list = [data_filepath for data_filepath in os.walk(top_path)]
+
+    tup = temp_list[0]
+    (dirpath, dirnames, filenames) = tup
+
+    cut_image_edge_size = 200
```
Comment (ronekko) [inline convnet.py]: image_sizeと重複してます．

File: convnet.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+        # 画像データの名前と拡張子を分離する
+        name, ext = os.path.splitext(filename)
+
+        # 3種類の訓練画像だけを指定する
+        text_type = name[5:6]
+        if(int(text_type) > 3):
+            print "filename:", filename
+            continue
+        image_y = image.shape[0]
```
Comment (ronekko) [inline convnet.py]: image_y → height

File: convnet.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        image_x = image.shape[1]
```
Comment (ronekko) [inline convnet.py]: image_x → width

File: convnet.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        y_p = y_select_point[0]
+        x_p = x_select_point[0]
+
+        image = image[y_p:y_p+image_size, x_p:x_p+image_size]
+        plt.imshow(image, cmap=plt.cm.gray)
+        plt.show()
+        plt.draw()
+        images.append(image)
+
+        x = np.array(images).reshape(-1, 1, image_size, image_size)
```
Comment (ronekko) [inline convnet.py]: これはforを抜けた後で．

File: convnet.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+        # １人あたり3種類の画像から1枚ずつ切り出し画像を作成する
+        x_select_points = width - image_size
+        y_select_points = heigh - image_size
+        x_select_point = np.random.permutation(x_select_points)
+        y_select_point = np.random.permutation(y_select_points)
+        y_p = y_select_point[0]
+        x_p = x_select_point[0]
+
+        image = image[y_p:y_p+image_size, x_p:x_p+image_size]
```
Comment (ronekko) [inline convnet.py]: image = image[y_p:y_p+image_size, x_p:x_p+image_size].copy()
にしたらどうなりますか？

File: convnet.py [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    file_numbers = []
+    for filename in filenames:
+        image = plt.imread(os.path.join(dirpath, filename))
+
+        # 画像データの名前と拡張子を分離する
+        name, ext = os.path.splitext(filename)
+
+        # 3種類の訓練画像だけを指定する
+        text_type = name[5:6]
+        if(int(text_type) > 3):
```
Comment (ronekko) [inline convnet.py]: ファイルを読み込んでしまった後でスキップするのは非効率的です．
読み込みはここより後に書いたほうが効率的です．

File: convnet.py [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+            train_loss, train_accuracy = loss_and_accuracy(model,
+                                                           x_batch_train,
+                                                           t_batch_train)
+            train_losses.append(train_loss.data)
+            train_accuracies.append(train_accuracy)
+        average_train_loss = np.array(train_losses).mean()
+        average_train_accuracy = np.array(train_accuracies).mean()
+        print "[train] Loss:", train_loss.data
+        print "[train] Accuracy:", train_accuracy
+        loss_history.append(train_loss.data.get())
```
Comment (ronekko) [inline convnet.py]: loss_history.append(average_train_loss)

File: convnet.py [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+        train_accuracy_history.append(train_accuracy)
```
Comment (ronekko) [inline convnet.py]: train_accuracy_history.append(average_train_accuracy)

File: convnet.py [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+            t_batch_train = cuda.to_gpu(t_train[batch_indexes])
+
+            train_loss, train_accuracy = loss_and_accuracy(model,
+                                                           x_batch_train,
+                                                           t_batch_train)
+            train_losses.append(train_loss.data)
+            train_accuracies.append(train_accuracy)
+        average_train_loss = np.array(train_losses).mean()
+        average_train_accuracy = np.array(train_accuracies).mean()
+        print "[train] Loss:", train_loss.data
```
Comment (ronekko) [inline convnet.py]: print "[train] Loss:", average_train_loss

File: convnet.py [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+        print "[train] Accuracy:", train_accuracy
```
Comment (ronekko) [inline convnet.py]: print "[train] Accuracy:", average_train_accuracy
